### PR TITLE
gh-104629: Build _testclinic extension module on Windows

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-104629

### CODE REVIEW

**Comment by erlend-aasland** [comment]
DISCLAIMER: I don't have a Windows dev environment set up; I don't know anything about Visual Code (Visual Studio?). I have no idea if this works.

FYI, I used random GUID's.

**Comment by AlexWaygood** [comment]
I know nothing about the build system, but I can confirm that this builds fine on my Windows machine (building from the command line), and `python -m test test_clinic` still passes. The number of skipped tests goes from 51 to 0 :)

**Comment by erlend-aasland** [comment]
> I know nothing about the build system, but I can confirm that this builds fine on my Windows machine (building from the command line), and `python -m test test_clinic -v` still passes. The number of skipped tests goes from 51 to 0 :)

Good to know! Thanks 😃

File: PCbuild/pcbuild.sln [hunk 1]
```diff
@@ -78,6 +79,8 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "_ssl", "_ssl.vcxproj", "{C6
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "_testcapi", "_testcapi.vcxproj", "{6901D91C-6E48-4BB7-9FEC-700C8131DF1D}"
 EndProject
+Project("{A840DDFB-ED50-484B-B527-B32E7CF90FD5}") = "_testclinic", "_testclinic.vcxproj", "{A840DDFB-ED50-484B-B527-B32E7CF90FD5}"
```
Comment (zooba) [inline PCbuild/pcbuild.sln]: This first guid should match the rest - it's the _kind_ of the project, not the specific identifier.

File: PCbuild/pcbuild.proj [hunk 2]
```diff
@@ -77,7 +77,7 @@
     <ExtensionModules Include="@(ExternalModules->'%(Identity)')" Condition="$(IncludeExternals)" />
     <Projects Include="@(ExtensionModules->'%(Identity).vcxproj')" Condition="$(IncludeExtensions)" />
     <!-- Test modules -->
-    <TestModules Include="_ctypes_test;_testbuffer;_testcapi;_testinternalcapi;_testembed;_testimportmultiple;_testmultiphase;_testsinglephase;_testconsole" />
+    <TestModules Include="_ctypes_test;_testbuffer;_testcapi;_testinternalcapi;_testembed;_testimportmultiple;_testmultiphase;_testsinglephase;_testconsole;_testclinic" />
```
Comment (zooba) [inline PCbuild/pcbuild.proj:80]: I believe there should be similar places to add the new module under `PC/layout` and `Tools/msi/test`, otherwise it won't get into the installers.

File: PCbuild/pcbuild.sln (hunk 1 above)
Comment (erlend-aasland) [reply PCbuild/pcbuild.sln]: Thanks, I'll update the PR.

File: PCbuild/pcbuild.proj (hunk 2 above)
Comment (erlend-aasland) [reply PCbuild/pcbuild.proj:80]: Seems `PC/layout` is handled by a glob:

https://github.com/python/cpython/blob/41de54378d54f7ffc38f07db4219e80f48c4249e/PC/layout/main.py#L33

For `Tools/msi/test` I'm kinda lost; I'm not sure what to do with those files :(

File: PCbuild/pcbuild.sln (hunk 1 above)
Comment (erlend-aasland) [reply PCbuild/pcbuild.sln:82]: How does 355f3397eb look?

Comment (erlend-aasland) [reply PCbuild/pcbuild.sln]: How does 355f3397eb2699a0b100b06f2b8ee3a927c65838 look?

Comment (erlend-aasland) [reply PCbuild/pcbuild.sln]: > How does [355f339](https://github.com/python/cpython/commit/355f3397eb2699a0b100b06f2b8ee3a927c65838) look?

Let me try again: how does 355f3397eb2699a0b100b06f2b8ee3a927c65838 _and_ e9aa5a3b668d63a270184074c66d881a2fd8e781 look?

File: PCbuild/pcbuild.proj (hunk 2 above)
Comment (zooba) [reply PCbuild/pcbuild.proj:80]: https://github.com/python/cpython/blob/main/Tools/msi/test/test_files.wxs#L3

Similar to pcbuild.proj, there's a macro that lists the names to generate. Just needs to be added there

Comment (erlend-aasland) [reply PCbuild/pcbuild.proj:80]: Tools/msi/test/test\_files updated in a3e7085f70a3e588d5f1b3fbb66f95a098bbdb3d

**Comment by erlend-aasland** [comment]
@zooba or @zware: I totally forgot about this PR. With the current relatively high volume of clinic changes going on, I think it is important that we land this. Could one of you check if my last changes are ok?

**Review by zware — APPROVED** [review]
Looks fine to me, and worked when I tried it.  If there's anything we've missed, it can be cleaned up later :)
